UT4MasterServer.Web/src/pages/ActivateAccount.vue

```diff
@@ -30,24 +30,21 @@ const accountService = new AccountService();
 async function activateAccount() {
   try {
     status.value = AsyncStatus.BUSY;
-    const { email, guid } = route.query;
+    const { accountId, guid } = route.query;
 
-    if (email?.toString() === '' || guid?.toString() === '') {
+    if (accountId?.toString() === '' || guid?.toString() === '') {
```

Comment:
You don't need to toString and compare to empty string here, just !accountId?.length will work. The query string vars are either string or string[] or null, so a length of 0 is empty string or empty array. This code actually wouldn't work if they were null or undefined, because undefined !== ''

Reply:
At first, it was giving me LocationQueryValue as variable type, that is why I had to do toString, but now when I check, destructuring an object to a new variable actually does converting to string.

**Example**: `/Login?activationLinkSent=done`
**Code**: 
```js
const { activationLinkSent: qActivationLinkSent } = route.query;

  console.log(
    'activationLinkSent:',
    activationLinkSent,
    qActivationLinkSent,
    qActivationLinkSent?.length
  );
```

**Output**: activationLinkSent: RefImpl {__v_isShallow: true, dep: Set(1), __v_isRef: true, _rawValue: false, _value: false} done 4

**Example 2**: `/Login?activationLinkSent=`
**Output**: activationLinkSent: RefImpl {__v_isShallow: true, dep: Set(1), __v_isRef: true, _rawValue: false, _value: false}  0

**Example 3**: `/Login?activationLinkSent`
**Output**: activationLinkSent: RefImpl {__v_isShallow: true, dep: Set(1), __v_isRef: true, _rawValue: false, _value: false} null undefined

I will what kind of check will cover all cases.

Reply:
You're getting that RefImpl because you're outputting activationLinkSent rather than activationLinkSent.value.

Reply:
The check should be !activationLinkSent?.length because that is just looking for a falsey value. Undefined is falsey and so is 0.

Reply:
But again, don't worry about it, I will fix it in the feature branch, you can merge this and I will pull and fix, it is easier to show you than explain in comments